hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/visitor/PlanCreateVisitor.java

```diff
@@ -253,6 +261,12 @@ public void onMapIndexScan(MapIndexScanPhysicalRel rel) {
         mapNames.add(table.getMapName());
     }
 
+    @Override
+    public void onSort(SortPhysicalRel rel) {
+        throw QueryException.error("Cannot perform ORDER BY clause without matching index."
```

Comment:
The error message doesn't explain the problem properly:
- Sorting could be located on top of any operator, and it could be impossible to add any index on it (imagine sorting on top of aggregation or join).
- It is not clear what `nested sorting` means, is it a well-known term in the SQL world? 
- If the user needs to add an index, can we tell him which exact tables and columns are affected?

Reply:
Improved in https://github.com/hazelcast/hazelcast/pull/18013/commits/e47c583eb9ed01ab607650bb398df31f90596ea0

Reply:
The error message is still incorrect. `ORDER BY` is supported on top of any operator, for so long the operator has the proper collation.  This collation, in turn, is pulled from children operators, and so on. Moreover, a map index scan below the `Sort` operator will almost never happen in a distributed setting. Last, but not least, our rels are private API, so exposing their names in the error message will not help user anyhow.

The ideal implementation will go through children operators and look for the original column in some map scan. There is some built-in support for this in Apache Calcite (`RelMdColumnOrigins`), but I do not know how reliable it is. Even if good enough, we would need to integrate the metadata propagation for our custom operators, which would like to consume a lot of time. Therefore, I would propose to have a simple and general enough error message for now and consider using the column origins for the future.

An example of error message that is general enough: `Cannot execute ORDER BY, because its input is not sorted. Consider adding a sorted index to the data source.`

Reply:
Returned back to the generic error message in https://github.com/hazelcast/hazelcast/pull/18013/commits/41414645f45a49f2b9d10951edbd4f3a2ea07a26